src/net/grunt/ClientLink.hpp

```diff
@@ -25,6 +25,16 @@ class Grunt::ClientLink : public WowConnectionResponse, Grunt::Pending, Grunt::T
             CMD_XFER_DATA                   = 49,
         };
 
+        enum STATE {
+            STATE_DISCONNECTED              = 0,
+            STATE_CONNECTING                = 1,
+            STATE_CANTCONNECT               = 2,
+            STATE_CONNECT_CHALLENGE         = 3,
+            STATE_CONNECT_VERSION           = 4,
+            STATE_RECONNECT_VERSION         = 5,
+            STATE_CONNECTED                 = 6
```

Comment:
I'm not sold on a few of these: in particular `STATE_CANTCONNECT` and `STATE_CONNECTING`. I don't have better names at the moment, but I'd suggest we take a little more cautious view here, and stick to something like this:

```cpp
STATE_DISCONNECTED              = 0,
STATE_GRUNT_1                   = 1,
STATE_GRUNT_2                   = 2,
STATE_GRUNT_3                   = 3,
STATE_CONNECT_VERSION           = 4,
STATE_RECONNECT_VERSION         = 5,
STATE_CONNECTED                 = 6
```

We can do some digging in 2.x and early 3.x builds and see if other debug strings shed light on the names for 1, 2, and 3.

Reply:
I figured that because state 2 is only referenced in some CantConnect function it would make sense to name it STATE_CANTCONNECT. The other state names also made sense to me when reversing them but I admit that I cannot be sure about most of them.

Reply:
It's used in `Grunt::ClientLink::WCConnected` (effectively the callback invoked after a connection occurs), too, which makes me think it's something a bit more nuanced than `CANTCONNECT`.

Reply:
I have compared the names to similar ones present in various builds and also checked what happens after the states are set like for example under GruntLogin::LogonResult and its LoginResponse::UpdateLoginStatus calls.

I believe the changes now bring the STATE enum to a very plausible... state. 😉